### PR TITLE
killDataSourceWhitelist in CoordinatorDynamicConfig accepts comma separated list of strings

### LINKED ISSUE #3055: killDataSourceWhitelist cannot be set using Coordinator UI
screen shot 2016-06-02 at 9 47 18 am
screen shot 2016-06-02 at 10 29 29 am

```
error:"Can not deserialize instance of java.util.HashSet out of VALUE_STRING token↵ at [Source: HttpInputOverHTTP@7522095a; line: 1, column: 220]"
```

### PR DESCRIPTION
**Pull request by himanshug**
… in addition to json array of strings so that coordinator console can do the updates correctly

fixes #3055

### CODE REVIEW

File: server/src/main/java/io/druid/server/coordinator/CoordinatorDynamicConfig.java [hunk 1]
```diff
@@ -58,7 +61,23 @@ public CoordinatorDynamicConfig(
...
+
+    // Strings are accepted in order to support update requests coming from coordinator console which can not send
+    // array of strings in the POST request.
+    // See https://github.com/druid-io/druid/issues/3055
+    if (killDataSourceWhitelist instanceof String) {
+      String[] tmp = ((String) killDataSourceWhitelist).split(",");
+      this.killDataSourceWhitelist = new HashSet<>();
+      for (int i = 0; i < tmp.length; i++) {
+        if (!tmp[i].isEmpty()) {
+          this.killDataSourceWhitelist.add(tmp[i]);
```
Comment (drcrallen) [inline server/src/main/java/io/druid/server/coordinator/CoordinatorDynamicConfig.java]: Should properry account for spaces after comma.

**Comment by drcrallen** [comment]
IMHO this should be handled at the UI and not in the service.

**Comment by gianm** [comment]
Does the UI already translate JSON arrays to csv when populating the text box? It'd need to do that for this fix to be effective.

**Comment by himanshug** [comment]
@drcrallen i would agree with you about fix to UI but there does not seem to be anyone available who understands and can fix it there.
@gianm I tested the patch, UI can do deserialization fine without any problems. at deserialization time, it "converts" array of strings to a comma separated string.

**Comment by gianm** [comment]
ok thanks for testing it @himanshug 

👍  from me. I agree it would be nicer to fix it in the UI, but I am okay with this faster fix for 0.9.1.

File: server/src/main/java/io/druid/server/coordinator/CoordinatorDynamicConfig.java (hunk 1 above)
Comment (himanshug) [inline server/src/main/java/io/druid/server/coordinator/CoordinatorDynamicConfig.java]: updated

**Comment by drcrallen** [comment]
As discussed in dev sync: Since the console is getting enhancement in general (soon-ish?) and this is a blocker for 0.9.1, I'm good with the proposed solution, but @himanshug can you please add a comment in the constructor as to why this is an object instead of a list? (probably just reference the github issue)

**Comment by himanshug** [comment]
@drcrallen added the comment.

**Comment by drcrallen** [comment]
cool, thanks 👍

**Comment by drcrallen** [comment]
Yay! it made it past the point where travis has been hanging recently.

**Pull request merged**